roles/wireguard/tasks/main.yml

```diff
@@ -0,0 +1,90 @@
+---
+- name: Install wireguard package
+  become: true
+  ansible.builtin.apt:
+    name: wireguard
+    state: present
+
+- name: Create public and private key - client
+  become: true
+  ansible.builtin.shell: |
+    set -o pipefail
+    umask 077
+    wg genkey | tee /etc/wireguard/client.key | wg pubkey > /etc/wireguard/client.pub
+  args:
+    creates: /etc/wireguard/client.key
+    executable: /bin/bash
+
+- name: Create public and private key - server
+  become: true
+  ansible.builtin.shell: |
+    set -o pipefail
+    umask 077
+    wg genkey | tee /etc/wireguard/server.key | wg pubkey > /etc/wireguard/server.pub
+  args:
+    creates: /etc/wireguard/server.key
+    executable: /bin/bash
+
+- name: Create preshared key
+  become: true
+  ansible.builtin.shell: umask 077; wg genpsk > /etc/wireguard/osism.psk
+  args:
+    creates: /etc/wireguard/osism.psk
+
+- name: Get preshared key
+  become: true
+  ansible.builtin.slurp:
+    src: /etc/wireguard/osism.psk
+  register: preshared_key
+
+- name: Get public key - client
+  become: true
+  ansible.builtin.slurp:
+    src: /etc/wireguard/client.pub
+  register: public_key_client
+
+- name: Get private key - client
+  become: true
+  ansible.builtin.slurp:
+    src: /etc/wireguard/client.key
+  register: private_key_client
+
+- name: Get public key - server
+  become: true
+  ansible.builtin.slurp:
+    src: /etc/wireguard/server.pub
+  register: public_key_server
+
+- name: Get private key - server
+  become: true
+  ansible.builtin.slurp:
+    src: /etc/wireguard/server.key
+  register: private_key_server
+
+- name: Copy wireguard-client.conf configuration file
+  ansible.builtin.template:
+    src: client.conf.j2
+    dest: "/home/{{ operator_user }}/wireguard-client.conf"
```

Comment:
This way we do not have to hardcode the /home/{{ operator_user}}. path.

```
# NOTE: http://stackoverflow.com/questions/33343215/how-to-get-remote-users-home-directory-in-ansible

- name: Get home directory of operator user
  become: true
  ansible.builtin.shell: |
    set -o pipefail
    getent passwd {{ operator_user }} | cut -d: -f6
  args:
    executable: /bin/bash
  changed_when: false
  register: operator_home
```

...

```
dest: "{{ result_create_operator_user.home }}/wireguard-client.conf"
```

Reply:
wouldn’t it be better to configure the operator home directory in a facts variable and in the operator repository?

Reply:
We really only need this on the manager. Therefore, I would not get this generically as a fact.

Reply:
Done